app/app.go

```diff
@@ -412,7 +413,27 @@ func (d *Deployments) handleArtifact(ctx context.Context,
 		}
 		return artifactID, errors.Wrap(err, "Fail to store the metadata")
 	}
-
+	i := 0
+	var updateTypes []string
+	if image.ArtifactMeta != nil {
+		updateTypes = make([]string, len(image.ArtifactMeta.Updates))
+		for _, t := range image.ArtifactMeta.Updates {
+			if t.TypeInfo.Type == nil {
+				continue
+			}
+			updateTypes[i] = *t.TypeInfo.Type
+			i++
+		}
+	}
+	if i > 0 {
+		err = d.db.SaveUpdateTypes(ctx, updateTypes[:i])
+		if err != nil {
+			l.Errorf(
+				"error while saving the update types for the artifact: %s",
+				err.Error(),
+			)
+		}
+	}
```

Comment:
```suggestion
	var updateTypes []string
	if image.ArtifactMeta != nil && len(image.ArtifactMeta.Updates) > 0 {
		updateTypes = make([]string, len(image.ArtifactMeta.Updates))
		for _, t := range image.ArtifactMeta.Updates {
			if t.TypeInfo.Type == nil {
				continue
			}
			updateTypes[i] = *t.TypeInfo.Type
		}
		if err = d.db.SaveUpdateTypes(ctx, updateTypes[:i]); err != nil {
			l.Errorf(
				"error while saving the update types for the artifact: %s",
				err.Error(),
			)
		}
	}
```

Reply:
sure, moved the second block into the if.